### PR TITLE
Backend registry

### PR DESCRIPTION
**Pull request by Speierers**
This PR seperated the pointer registry datastructures for the LLVM and CUDA backends.

### CODE REVIEW

File: src/registry.cpp [hunk 1]
```diff
@@ -251,38 +284,52 @@ void jitc_registry_set_attr(void *ptr, const char *name,
...
-            CUresult ret = cuMemAllocManaged((CUdeviceptr *) &ptr, new_size, CU_MEM_ATTACH_GLOBAL);
+            CUresult ret = cuMemAlloc((CUdeviceptr *) &ptr, new_size);
             if (ret != CUDA_SUCCESS) {
                 jitc_malloc_trim(true, true);
-                cuda_check(cuMemAllocManaged((CUdeviceptr *) &ptr, new_size, CU_MEM_ATTACH_GLOBAL));
+                cuda_check(cuMemAlloc((CUdeviceptr *) &ptr, new_size));
             }
-            cuda_check(cuMemAdvise((CUdeviceptr) ptr, new_size, CU_MEM_ADVISE_SET_READ_MOSTLY, 0));
+
+            // cuda_check(cuCtxSynchronize());
```
Comment (wjakob) [inline src/registry.cpp:295]: This can be removed now?

File: src/vcall.cpp [hunk 2]
```diff
@@ -246,7 +246,7 @@ void jitc_var_vcall(const char *name, uint32_t self, uint32_t mask,
 
         for (uint32_t j = se_offset[i]; j != se_offset[i + 1]; ++j)
             jitc_var_vcall_collect_data(vcall->data_map, data_size, i,
-                                        ts->side_effects[j]);
+                                        vcall->se[j - se_offset[0]]);
```
Comment (wjakob) [inline src/vcall.cpp:249]: Should this go into a separate commit?

**Review by wjakob — APPROVED** [review]
Looks great, thanks -- only two minor comments.

File: src/vcall.cpp (hunk 2 above)
Comment (Speierers) [reply src/vcall.cpp:249]: yes!

File: src/registry.cpp (hunk 1 above)
Comment (Speierers) [reply src/registry.cpp:295]: Haven't tried this on Windows yet. But it should be yes
